api/api.go

```diff
@@ -359,15 +360,33 @@ func (api *API) delSilence(w http.ResponseWriter, r *http.Request) {
 }
 
 func (api *API) listSilences(w http.ResponseWriter, r *http.Request) {
-	sils, err := api.silences.All()
+	var (
+		num    = r.FormValue("n")
+		offset = r.FormValue("offset")
```

Comment:
You're right of course, we can't have the best of both worlds. What I described doesn't work for pagination by showing page numbers. Let's continue with offset then.